src/syntax.ml

```diff
@@ -671,7 +674,15 @@ let rec block_element_list :
 
     match peek input with
     (* Terminators: the two tokens that terminate anything. *)
-    | ({ value = `End; _ } | { value = `Right_brace; _ }) as next_token -> (
+    | { value = `End; _ } as next_token -> (
+        match context with
+        | Top_level -> (List.rev acc, next_token, where_in_line)
+        | In_shorthand_list -> (List.rev acc, next_token, where_in_line)
+        | In_explicit_list -> (List.rev acc, next_token, where_in_line)
+        | In_tag -> (List.rev acc, next_token, where_in_line)
+        | In_table_cell -> (List.rev acc, next_token, where_in_line)
+        | In_code_results -> (List.rev acc, next_token, where_in_line))
```

Comment:
Most cases have the same type and could perhaps be written with an or-pattern ?

Reply:
Annoyingly not (the return types are different in each case). See the comment immediately below.

Reply:
Indeed.